images/virtualization-artifact/pkg/controller/vm/internal/validators/sizing_policy_compliency_validator.go

```diff
@@ -0,0 +1,231 @@
+	if sp.CoreFractions == nil {
+		return
+	}
+
+	fractionStr := strings.ReplaceAll(vm.Spec.CPU.CoreFraction, "%", "")
+	fraction, err := strconv.Atoi(fractionStr)
+	if err != nil {
+		errorsArray = append(errorsArray, fmt.Errorf("cpu fraction value is invalid"))
+		return
+	}
+
+	hasInSizePolicyFractions := false
+	for _, spFraction := range sp.CoreFractions {
+		if fraction == int(spFraction) {
+			hasInSizePolicyFractions = true
+		}
+	}
+
+	if !hasInSizePolicyFractions {
+		errorsArray = append(errorsArray, fmt.Errorf("vm core fraction incorrect"))
+	}
+
+	return
+}
+
+func validateVMMemory(vm *v1alpha2.VirtualMachine, sp *v1alpha2.SizingPolicy) (errorsArray []error) {
+	if sp.Memory == nil || sp.Memory.Max.IsZero() {
+		return
+	}
+
+	if vm.Spec.Memory.Size.Cmp(sp.Memory.Min) == -1 {
+		errorsArray = append(errorsArray, fmt.Errorf(
+			"requested VM memory (%s) lesser than valid minimum, available range [%s, %s]",
+			vm.Spec.Memory.Size.String(),
+			sp.Memory.Min.String(),
+			sp.Memory.Max.String(),
+		))
+	}
+
+	if vm.Spec.Memory.Size.Cmp(sp.Memory.Max) == 1 {
+		errorsArray = append(errorsArray, fmt.Errorf(
+			"requested VM memory (%s) greater than valid maximum, available range [%s, %s]",
+			vm.Spec.Memory.Size.String(),
+			sp.Memory.Min.String(),
+			sp.Memory.Max.String(),
+		))
+	}
+
+	if !sp.Memory.Step.IsZero() {
+		err := checkInGrid(vm.Spec.Memory.Size, sp.Memory.Min, sp.Memory.Max, sp.Memory.Step, "VM memory")
+		if err != nil {
+			errorsArray = append(errorsArray, err)
+		}
+	}
+
+	return
+}
+
+func validatePerCoreMemory(vm *v1alpha2.VirtualMachine, sp *v1alpha2.SizingPolicy) (errorsArray []error) {
+	if sp.Memory == nil || sp.Memory.PerCore.Max.IsZero() {
+		return
+	}
+
+	// not have a default dividing :(
+	// dirty, I know
+	// wash your hands after read this
+	vmPerCore := vm.Spec.Memory.Size.Value() / int64(vm.Spec.CPU.Cores)
+	perCoreMemory := resource.NewQuantity(vmPerCore, resource.BinarySI)
+
+	if perCoreMemory.Cmp(sp.Memory.PerCore.Min) == -1 {
+		errorsArray = append(errorsArray, fmt.Errorf(
+			"requested VM per core memory (%s) lesser than valid minimum, available range [%s, %s]",
+			perCoreMemory.String(),
+			sp.Memory.PerCore.Min.String(),
+			sp.Memory.PerCore.Max.String(),
+		))
+	}
+
+	if perCoreMemory.Cmp(sp.Memory.PerCore.Max) == 1 {
+		errorsArray = append(errorsArray, fmt.Errorf(
+			"requested VM per core memory (%s) greater than valid maximum, available range [%s, %s]",
+			perCoreMemory.String(),
+			sp.Memory.PerCore.Min.String(),
+			sp.Memory.PerCore.Max.String(),
+		))
+	}
+
+	if !sp.Memory.Step.IsZero() {
+		err := checkInGrid(*perCoreMemory, sp.Memory.PerCore.Min, sp.Memory.PerCore.Max, sp.Memory.Step, "VM per core memory")
+		if err != nil {
+			errorsArray = append(errorsArray, err)
+		}
+	}
+
+	return
+}
+
+func checkInGrid(value, min, max, step resource.Quantity, source string) (err error) {
```

Comment:
It is not a "grid", grid is something with at lease 2 dimensions, it is "ranges" or "buckets".

Reply:
Also, add comment with the brief algorithm description, as equality with integers (-1, 0, 1) is not descriptive.